File: Server/Entities/Notification.cs
using SubTrack.Notifications.Enums;

namespace SubTrack.Notifications.Entities
{
    public class Notification
    {
        public Guid Id { get; set; }

        public Guid UserId { get; set; }

        public string Title { get; set; } = string.Empty;

        public string Message { get; set; } = string.Empty;

        public NotificationType Type { get; set; } = NotificationType.General;

        public bool IsRead { get; set; }

        public DateTime CreatedAt { get; set; }
    }
}

File: Server/src/main/java/com/subtrack/entity/Notification.java
package com.subtrack.entity;
import jakarta.persistence.*; import java.time.Instant; import java.util.UUID;
@Entity @Table(name="Notification")
public class Notification {
@Id @GeneratedValue(strategy=GenerationType.UUID) private UUID id;
@Column(name="user_id",nullable=false) private UUID userId;
private String title; @Column(length=2000) private String message;
@Enumerated(EnumType.STRING) private NotificationType type=NotificationType.General;
private boolean isRead=false; private Instant createdAt=Instant.now();
protected Notification(){}
public Notification(UUID userId,String title,String message,NotificationType type){this.userId=userId;this.title=title;this.message=message;this.type=type==null?NotificationType.General:type;}
public UUID getId(){return id;} public UUID getUserId(){return userId;} public String getTitle(){return title;} public String getMessage(){return message;} public NotificationType getType(){return type;} public boolean isRead(){return isRead;} public Instant getCreatedAt(){return createdAt;} public void markRead(){isRead=true;}
}